controllers/cloud.redhat.com/makers/maker.go

```diff
@@ -320,6 +340,98 @@ func initJob(app *crd.ClowdApp, env *crd.ClowdEnvironment, cj *batch.CronJob, nn
 	cj.Spec.JobTemplate.Spec.Template.SetAnnotations(annotations)
 }
 
+func applyOneShotJob(app *crd.ClowdApp, env *crd.ClowdEnvironment, j *batchv1.Job, nn types.NamespacedName, job crd.Job, hash string) {
+	labels := app.GetLabels()
+	labels["pod"] = nn.Name
+	app.SetObjectMeta(j, crd.Name(nn.Name), crd.Labels(labels))
+
+	pod := job.PodSpec
+
+	// https://kubernetes.io/docs/concepts/workloads/controllers/job/#specifying-your-own-pod-selector
+	// This is dumb, stupid, and bad. Jobs will not apply due to an error
+	// with the UID of the Job Controller. Needs more research.
+	var manual bool = true
+	j.Spec.Selector = &metav1.LabelSelector{}
```

Comment:
It looks as if the CronJob has the Job spec nested under the JobTemplate, if this is the case, we can likely utilise the ame code and make this much cleaner. I didn't look into it in detail - but something worth looking in to.